components/hab/src/cli.rs

```diff
@@ -264,6 +264,46 @@ pub fn get() -> App<'static, 'static> {
                     (@arg AUTH_TOKEN: -z --auth +takes_value "Authentication token for Builder")
                 )
             )
+            (@subcommand secret =>
+                (about: "Commands related to secret management")
+                (@setting ArgRequiredElseHelp)
+                (@subcommand generate => 
```

Comment:
The name `generate` implies to me that I don't have to come up with the value. To me `add` is the more natural counterpart to `delete`.

Reply:
I talked with @ryankeairns about this and he wanted the UX to be consistent with `origin key generate`. Previously I had `origin secret create` to follow the CRUD pattern.

Reply:
I 💯 on-board with consistency, but in this case I'd say it's a different operation and so should have a different name to make that clear. Not a hill I'm looking to die on or anything, it just struck me as unexpected. I'm interested what @ryankeairns thinks.

Reply:
Just so I'm following... generate generates a new key; are we also creating/generating a new secret?

Reply:
@elliott-davis can correct me if I'm wrong, but what we're doing here is asking builder to store an encrypted key/value pair that we are supplying. Builder isn't generating a value for us.

Reply:
I don't want to beat this to death (too late!), but perhaps `upload` is the verb then.